website/content/docs/internals/filesystem.mdx

```diff
@@ -42,8 +42,15 @@ allocation directory like the one below.
   log shipper. This is the directory that's provided to the task as the
   `NOMAD_ALLOC_DIR`. Note that this `alloc/` directory is not the same as the
   "allocation working directory", which is the top-level directory. All tasks
-  in a task group can read and write to the `alloc/` directory. Within the
-  `alloc/` directory are three standard directories:
+  in a task group can read and write to the `alloc/` directory, but the path
+  may differ depending on the task driver's [filesystem isolation mode]. Tasks
+  with driver isolation of [`none`](#none-isolation) may reference this
+  directory using the `NOMAD_ALLOC_DIR` environment variable or a relative path
+  that traverses up into the allocation working directory (`../alloc/`). Other
+  task drivers will have this directory provided within their working
+  directory, and may access it via the `alloc/` path, such as with
+  [`image`](#image-isolation) and [`chroot`](#chroot-isolation) isolations.
+  Within the `alloc/` directory are three standard directories:
```

Comment:
This is all strictly true, but tasks should _always_ refer to the `NOMAD_ALLOC_DIR` env var provided and not rely on the specific implementation of the driver (which is why this section says "This is the directory that's provided to the task as the `NOMAD_ALLOC_DIR`" just above here). The details on how paths look from inside the task is discussed in the sections for each filesystem isolation mode further down.

If one wanted to change this, I'd probably just call that out specifically:

```suggestion
  in a task group can read and write to the `alloc/` directory. But the full host 
  path may differ depending on the task driver's [filesystem isolation mode], so
  tasks should always used the `NOMAD_ALLOC_DIR` environment variable.
  to find this path rather than relying on the specific implementation of the 
  [`none`](#none-isolation), [`chroot`](#chroot-isolation), or [`image`](#image-isolation) 
  modes. Within the `alloc/` directory are three standard directories:
```

Reply:
Thanks for the input @tgross! I agree with you re. implementation details, and I like the suggestion. 

What do you think @jmwilkinson? Does this clarify things enough?